src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgres.java

```diff
@@ -433,15 +442,91 @@ private void mkdirs(File dir)
     private static final AtomicReference<File> BINARY_DIR = new AtomicReference<>();
     private static final Lock PREPARE_BINARIES_LOCK = new ReentrantLock();
 
-    private File prepareBinaries(PgBinaryResolver pgBinaryResolver) {
+    /**
+     * Get current operating system string. The string is used in the appropriate postgres binary name.
+     *
+     * @return Current operating system string.
+     */
+    private static String getOS(){
+        return SystemUtils.IS_OS_WINDOWS ? "Windows" : system("uname", "-s").get(0);
```

Comment:
I didn't want to redesign existing functionality / naming and I am not able to test what Apple returns.